src/ConnectedMode/Binding/ImportBeforeFileGenerator.cs

```diff
@@ -21,77 +21,79 @@
-        }
-
-        private void WriteFile(string xml)
-        {
+            var resource = ReadResourceFile();
             var pathToImportBefore = GetPathToImportBefore();
+            var fullPath = Path.Combine(pathToImportBefore, targetsFileName);
 
-            if (!fileSystem.Directory.Exists(pathToImportBefore))
+            try
             {
-                fileSystem.Directory.CreateDirectory(pathToImportBefore);
+                if (!fileSystem.Directory.Exists(pathToImportBefore))
+                {
+                    logger.LogVerbose(Resources.ImportBeforeFileGenerator_CreatingDirectory, pathToImportBefore);
+                    fileSystem.Directory.CreateDirectory(pathToImportBefore);
+                }
+
+                if (fileSystem.File.Exists(fullPath) && fileSystem.File.ReadAllText(fullPath) == resource)
+                {
+                    logger.LogVerbose(Resources.ImportBeforeFileGenerator_FileAlreadyExists);
+                    return;
+                }
+
+                logger.LogVerbose(Resources.ImportBeforeFileGenerator_WritingTargetFileToDisk);
+                fileSystem.File.WriteAllText(fullPath, resource);
             }
-
-            var fullPath = Path.Combine(pathToImportBefore, targetsFileName);
-
-            if (fileSystem.File.Exists(fullPath) && fileSystem.File.ReadAllText(fullPath) == xml)
+            catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
             {
-                return;
+                logger.WriteLine(Resources.ImportBeforeFileGenerator_FailedToWriteFile);
+                logger.LogVerbose(Resources.ImportBeforeFileGenerator_FailedToWriteFile_Verbose, ex.Message);
             }
-
-            fileSystem.File.WriteAllText(fullPath, xml);
         }
 
-        private XmlWriter Create(StringBuilder sb)
+        private string ReadResourceFile()
         {
-            var stringWriter = new StringWriter(sb);
+            logger.LogVerbose(Resources.ImportBeforeFileGenerator_LoadingResourceFile);
 
-            var settings = new XmlWriterSettings
+            var resourcePath = "SonarLint.VisualStudio.ConnectedMode.Embedded.SonarLintTargets.txt";
+
+            using (var stream = GetType().Assembly.GetManifestResourceStream(resourcePath))
             {
-                ConformanceLevel = ConformanceLevel.Fragment,
-                Encoding = Encoding.UTF8,
-                OmitXmlDeclaration = true,
-                Indent = true,
-                CloseOutput = true,
-                WriteEndDocumentOnClose = true
-            };
-
-            return XmlWriter.Create(stringWriter, settings);
+                if (stream != null)
+                {
+                    using (var reader = new StreamReader(stream))
+                    {
+                        var data = reader.ReadToEnd();
+                        return data;
+                    }
+                }
+            }
+
+            return "";
```

Comment:
I am not actually handling this case, and I am unsure if I should. Can I assume this will not fail considering it is our own text file? Or should I check IfNullOrEmpty?

Reply:
I thought `GetManifestResourceStream` throws if it can't find the specified resource.
If that is the case, your simpler version of the code in the comment above would be better.